narwhals/_pandas_like/namespace.py

```diff
@@ -130,11 +130,17 @@ def all(self) -> PandasLikeExpr:
         )
 
     def lit(self, value: Any, dtype: dtypes.DType | None) -> PandasLikeExpr:
+        if self._implementation is Implementation.CUDF:
+            import cupy as np  # ignore-banned-import
```

Comment:
cudf depends on cupy, and cupy implements [full](https://docs.cupy.dev/en/stable/reference/generated/cupy.full.html)